lib/cretonne/src/dominator_tree.rs

```diff
@@ -323,6 +329,77 @@ impl DominatorTree {
     }
 }
 
+impl DominatorTree {
+    /// When splitting an `Ebb` using `Layout::split_ebb`, you can use this method to update
+    /// the dominator tree locally rather than recomputing it.
+    ///
+    /// `old_ebb` is the `Ebb` before splitting, and `new_ebb` is the `Ebb` which now contains
+    /// the second half of `old_ebb`. `split_jump_inst` is the terminator jump instruction of
+    /// `old_ebb` that points to `new_ebb`.
+    pub fn recompute_split_ebb(&mut self, old_ebb: Ebb, new_ebb: Ebb, split_jump_inst: Inst) {
+        if !self.is_reachable(old_ebb) {
+            // old_ebb is unreachable, it stays so and new_ebb is unreachable too
+            *self.nodes.ensure(new_ebb) = DomNode {
+                rpo_number: 0,
+                idom: Some(split_jump_inst).into(),
+            };
```

Comment:
Early return + unindent.

I don't think we should store an idom for unreachable nodes. Just set it to `None`:

```rust
*self.nodes.ensure(new_ebb) = Default::default();
```